app/routes/city-council-districts.$cityCouncilDistrictId.capital-projects_.$managingCode.$capitalProjectId.tsx

```diff
@@ -0,0 +1,65 @@
+import { LoaderFunctionArgs, json } from "@remix-run/node";
+import {
+  useLoaderData,
+  useNavigate,
+  useParams,
+  useSearchParams,
+} from "@remix-run/react";
+import {
+  findCapitalProjectByManagingCodeCapitalProjectId,
+  findAgencies,
+} from "../gen";
+import { CapitalProjectDetailPanel } from "../components/CapitalProjectDetailPanel";
+import { useState } from "react";
+
+export async function loader({ params }: LoaderFunctionArgs) {
+  const { managingCode, capitalProjectId } = params;
+  const agenciesResponse = await findAgencies({
+    baseURL: `${import.meta.env.VITE_ZONING_API_URL}/api`,
+  });
+  if (managingCode === undefined || capitalProjectId === undefined) {
+    throw json("Bad Request", { status: 400 });
+  }
+  const capitalProjectPromise =
+    findCapitalProjectByManagingCodeCapitalProjectId(
+      managingCode,
+      capitalProjectId,
+      {
+        baseURL: `${import.meta.env.VITE_ZONING_API_URL}/api`,
+      },
+    );
+
+  const [capitalProject] = await Promise.all([capitalProjectPromise]);
+
+  return json({
+    capitalProject,
+    agencies: agenciesResponse.agencies,
+  });
+}
+
+export default function CapitalProjectByCityCouncilDistrictId() {
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+  const { cityCouncilDistrictId } = useParams();
+  const { capitalProject, agencies } = useLoaderData<typeof loader>();
+  const [panelIsFullSize, setPanelIsFullSize] = useState(false);
+
+  const togglePanelIsFullSize = () =>
+    setPanelIsFullSize((panelIsFullSize) => !panelIsFullSize);
+
+  return (
+    <CapitalProjectDetailPanel
```

Comment:
If we're changing the component name from `CapitalProjectDetailPanel` to `CapitalProjectDetail` we should stay consistent with it

Reply:
I think the way I have been defining and exporting "main" components from their folders might be causing some confusion here.

The top level component is still `CapitalProjectDetailPanel`. However, it is exported from the `index` file of the `CapitalProjectDetailPanel` folder. 

`CapitalProjectDetail` is a sub-component of the `CapitalProjectDetailPanel`. It is  focuses specifically on the `Details` of the project and used only within the `CapitalProjectDetailPanel`.

There is whole discussion around the merits of defining components within the `index` folder. I don't really know how we want to have that discussion.